x/chainlink/keeper/keeper_test.go

```diff
@@ -294,13 +296,359 @@ func TestKeeper_GetLatestRoundId(t *testing.T) {
+		AssignerAddress: acc,
+	})
+
+	// check if module owner doesn't exist anymore
+	data := moduleStore.Get(types.GetModuleOwnerKey(acc.String()))
+	require.Equal(t, []byte(nil), data)
 }
 
 func TestKeeper_GetModuleOwnerList(t *testing.T) {
+	k, ctx := setupKeeper(t)
+
+	_, pubKey1, acc1 := testdata.KeyTestPubAddr()
+	cosmosPubKey1, err := sdk.Bech32ifyPubKey(sdk.Bech32PubKeyTypeAccPub, pubKey1)
+	require.NoError(t, err)
+	owner1 := &types.MsgModuleOwner{
+		Address: acc1,
+		PubKey:  []byte(cosmosPubKey1),
+	}
+
+	_, pubKey2, acc2 := testdata.KeyTestPubAddr()
+	cosmosPubKey2, err := sdk.Bech32ifyPubKey(sdk.Bech32PubKeyTypeAccPub, pubKey2)
+	require.NoError(t, err)
+	owner2 := &types.MsgModuleOwner{
+		Address: acc2,
+		PubKey:  []byte(cosmosPubKey2),
+	}
+
+	testCases := []struct {
+		test        string
+		moduleOwner *types.MsgModuleOwner
+		expected    []*types.MsgModuleOwner
+	}{
+		{
+			test:        "owner 1",
+			moduleOwner: owner1,
+			expected:    []*types.MsgModuleOwner{owner1},
+		},
+		{
+			test:        "owner 2 and previous one",
+			moduleOwner: owner2,
+			expected:    []*types.MsgModuleOwner{owner1, owner2},
+		},
+	}
+
+	// Set module owner and try retrieve it
+	for _, tc := range testCases {
+		t.Run(tc.test, func(t *testing.T) {
+			k.SetModuleOwner(ctx, tc.moduleOwner)
+
+			moduleOwner := k.GetModuleOwnerList(ctx)
+			require.Equal(t, len(tc.expected), len(moduleOwner.GetModuleOwner()))
+		})
+	}
+}
+
+func TestKeeper_SetFeed(t *testing.T) {
+	k, ctx := setupKeeper(t)
+	feedInfoStore := ctx.KVStore(k.feedInfoStoreKey)
+
+	feedToInsert := types.MsgFeed{
+		FeedId:    "feed1",
+		FeedOwner: GenerateAccount(),
+		DataProviders: types.DataProviders{
+			{Address: GenerateAccount()},
+			{Address: GenerateAccount()},
+		},
+		SubmissionCount:           1,
+		HeartbeatTrigger:          2,
+		DeviationThresholdTrigger: 3,
+		FeedReward: &types.FeedRewardSchema{
+			Amount:   4,
+			Strategy: "none",
+		},
+		Desc:               "desc1",
+		ModuleOwnerAddress: GenerateAccount(),
+	}
+
+	k.SetFeed(ctx, &feedToInsert)
+
+	data := feedInfoStore.Get(types.GetFeedInfoKey(feedToInsert.GetFeedId()))
+	var feed types.MsgFeed
+	err := k.cdc.UnmarshalBinaryBare(data, &feed)
+	require.NoError(t, err)
+
+	require.Equal(t, feedToInsert.GetFeedId(), feed.GetFeedId())
+	require.Equal(t, feedToInsert.GetFeedOwner(), feed.GetFeedOwner())
+	require.EqualValues(t, feedToInsert.GetDataProviders(), feed.GetDataProviders())
+	require.Equal(t, feedToInsert.GetSubmissionCount(), feed.GetSubmissionCount())
+	require.Equal(t, feedToInsert.GetHeartbeatTrigger(), feed.GetHeartbeatTrigger())
+	require.Equal(t, feedToInsert.GetDeviationThresholdTrigger(), feed.GetDeviationThresholdTrigger())
+	require.Equal(t, feedToInsert.GetFeedReward(), feed.GetFeedReward())
+	require.Equal(t, feedToInsert.GetDesc(), feed.GetDesc())
+	require.Equal(t, feedToInsert.GetModuleOwnerAddress(), feed.GetModuleOwnerAddress())
+}
+
+func TestKeeper_GetFeed(t *testing.T) {
```

Comment:
looks like a carbon copy of the above test. should we just name the test as `TestKeeper_GetSetFeed` and combine the two?

Reply:
You right, on SetFeed, I directly retrieve data from the store and on GetFeed, I use the keeper method
Since it's the same code in the backend, I'm going to merge the two tests ;)